Plate_2_data/4.processing_features/scripts/2.combine_sc_runs_plate2.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Combine both SHSY5Y run parquet files into one
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pandas as pd
+import pathlib
+
+from pycytominer.cyto_utils import output
+
+
+# ## Set paths
+
+# In[2]:
+
+
+# set path to parquet directory
+parquet_dir = pathlib.Path("./data")
+
+# set paths to each individual run file after annotation
+first_run_sc_path = pathlib.Path(f"{parquet_dir}/SHSY5Y_first_run_sc.parquet")
+second_run_sc_path = pathlib.Path(f"{parquet_dir}/SHSY5Y_second_run_sc.parquet")
+
+# set path for combined run parquet file
+merged_runs_path = pathlib.Path(f"{parquet_dir}/SHSY5Y_sc.parquet")
+
+
+# ## Combine the parquet files into one
+
+# In[3]:
+
+
+# read parquet files into pandas dataframes
+first_run_sc = pd.read_parquet(first_run_sc_path)
+second_run_sc = pd.read_parquet(second_run_sc_path)
+
+# concatenate dataframes and save as parquet file
+SHSY5Y_run_sc = pd.concat([first_run_sc, second_run_sc], ignore_index=True)
```

Comment:
If possible, consider using `parquet_dir` to read these files and perform an implicit concatenation instead of reading the individual files manually (`pd.read_parquet` allows single file and multi-file parquet dataset reads).

Note: when providing a directory in this way, all parquet files may be concatenated together (which may be more than intended). Because of this, it may be beneficial to create subdirectories under `./data` to make the files more directly "concattable" using the directory method. Doing this might help organize the data as well as reducing the amount of code needed.

Reply:
Would this work by using `pd.read_parquet(path_to_directory)`? Then the resulting Dataframe is the combined parquet files?

@d33bs

Reply:
Yes that's correct! Here's a quick example to think about when designing things using parquet directories and Pandas:

```python
import pathlib

import pandas as pd

# create example data directories
pathlib.Path("./data/dataset_1").mkdir(parents=True, exist_ok=True)
pathlib.Path("./data/dataset_2").mkdir(parents=True, exist_ok=True)

# export data to ./data/dataset_1
pd.DataFrame({"col_1": [1, 2, 3], "col_2": [1, 2, 3]}).to_parquet(
    "./data/dataset_1/table_1.parquet"
)
pd.DataFrame({"col_1": [4, 5, 6], "col_2": [4, 5, 6]}).to_parquet(
    "./data/dataset_1/table_2.parquet"
)

# export data to ./data/dataset_2
pd.DataFrame({"col_a": [1, 2, 3], "col_b": [1, 2, 3]}).to_parquet(
    "./data/dataset_2/table_1.parquet"
)
pd.DataFrame({"col_a": [4, 5, 6], "col_b": [4, 5, 6]}).to_parquet(
    "./data/dataset_2/table_2.parquet"
)

# read files under ./data/dataset_1
print(pd.read_parquet("./data/dataset_1"))

# read files under ./data/dataset_2
print(pd.read_parquet("./data/dataset_2"))

# be careful - mixing files with differing schema
# this way may have unexpected results!
print(pd.read_parquet("./data"))
```

Reply:
Based on discussion yesterday, I will keep the way it is now but keep this in mind for the future!